src/implementations/svd.jl

```diff
@@ -203,7 +242,39 @@ function svd_trunc!(A::AbstractMatrix, USVᴴ, alg::TruncatedAlgorithm)
     return truncate!(svd_trunc!, USVᴴ′, alg.trunc)
 end
 
-### GPU logic
+# Diagonal logic
+# --------------
+function svd_full!(A::AbstractMatrix, USVᴴ, alg::DiagonalAlgorithm)
+    check_input(svd_full!, A, USVᴴ, alg)
+    Ad = diagview(A)
+    U, S, Vᴴ = USVᴴ
+    Sd = diagview(S)
+    Sd .= abs.(Ad)
+    p = sortperm(Sd; rev=true)
+    permute!(Sd, p)
+    T = eltype(Vᴴ)
+    zero!(U)
+    zero!(Vᴴ)
+    @inbounds for (i, pi) in enumerate(p)
```

Comment:
The gauge fixing for the svd in the general matrix case is such that the largest element on every column of `U` is positive, and the sign/phase is absorved in `Vᴴ`. So, `U` should be a pure permutation matrix, and `Vᴴ` should incorporate the sign of the diagonal element.

Reply:
Once again, an implementation without scalar indexing would be great, but in this case I would also need to think about how to do that.